cfgmgr/buffer_check_headroom_barefoot.lua

```diff
@@ -0,0 +1,17 @@
+-- KEYS - port name
+
+local port = KEYS[1]
+local state_db = "6"
+local ret = {}
+
+-- Connect to STATE_DB
+redis.call('SELECT', state_db)
+local max_headroom_size = tonumber(redis.call('HGET', 'BUFFER_MAX_PARAM_TABLE|' .. port, 'max_headroom_size'))
+if max_headroom_size == 0 or max_headroom_size == nil then
```

Comment:
FYI.
`BUFFER_MAX_PARAM_TABLE|<port>|max_headroom_size` is the maximum accumulative headroom that can be applied to a port. It will always be exposed to `STATE_DB` during orchagent initialization as long as being supported by the vendor SAI. It won't be changed after initialization. It is not relevant to whether the shared headroom pool is enabled.
So In case the shared headroom pool is a must on Barefoot's platforms, I would like to suggest enabling it in `buffers_config.j2` in `DEFAULT_LOSSLESS_BUFFER_PARAMETER` and always returning true here.